test/replcompletions.jl

```diff
@@ -390,6 +390,23 @@ c, r, res = test_scomplete(s)
         @test s[r] == "tmpfoob"
         rm(dir)
     end
+
+    # Tests detecting of files in the env path (in shell mode)
+    let
+        oldpath = ENV["PATH"]
+        ENV["PATH"] = homedir()
+        file = joinpath(path, "tmp-executable")
```

Comment:
Assuming you are copying the block above, I think the `path` should be `homedir()`.

You also don't need to do this in the home directory, #11440 need it because it is testing the `~` expansion. (Although a hidden and random directory might be better @blakejohnson ). You should use a `tempdir` for this PR.